modules/Emails/Email.php

```diff
@@ -3489,7 +3489,13 @@ public function fill_in_additional_detail_fields()
             $this->status_name = $app_list_strings['dom_email_status'][$this->status];
         }
 
-        if (empty($this->name) && empty($_REQUEST['record']) && !empty($mod_strings['LBL_NO_SUBJECT'])) {
+        if($_REQUEST['targetModule'] === "Cases" && !empty($_REQUEST["ids"])) {
+            $caseId = $_REQUEST['ids'];
+            $aCase = BeanFactory::getBean('Cases', $caseId);
+
+            $this->name = str_replace('%1', $aCase->case_number, $aCase->getEmailSubjectMacro() . " " . $aCase->name);
```

Comment:
Hi @cameronblaikie,

Could we add some checks here please? I was thinking on the following ones:
- Check that `$_REQUEST['ids']` is a string and not an array
- Check if `$aCase = BeanFactory::getBean('Cases', $caseId);` returned a bean or not.